GAE/src/org/waterforpeople/mapping/app/web/dto/DeviceNotificationRequest.java

```diff
@@ -16,9 +16,29 @@
 
 package org.waterforpeople.mapping.app.web.dto;
 
+import javax.servlet.http.HttpServletRequest;
+
 public class DeviceNotificationRequest extends LocationBeaconRequest {
 
     private static final long serialVersionUID = 287773188257035166L;
+    private static final String SURVEYS_PARAM = "formId";
+    
+    private String[] surveyIds;
+    
+    public String[] getSurveyIds() {
+        return surveyIds;
+    }
+
+    public void setSurveyIds(String[] surveyIds) {
+        this.surveyIds = surveyIds;
```

Comment:
If the parameter is passed a `String[]` here is where we should do the cast to `List<Long>`